lib/index.js

```diff
@@ -17,12 +24,30 @@ Controller.prototype.setTarget = function(target) {
   this.target = target;
 };
 
-Controller.prototype.update = function(current_value) {
-  this.current_value = current_value;
-
-  var error = (this.target - this.current_value);
-  this.sumError = this.sumError + error;
+Controller.prototype.update = function(currentValue) {
+  this.currentValue = currentValue;
+
+  // Calculate dt
+  var dt = this.dt;
+  if (this.dt === true) {
+    var currentTime = Date.now();
+    if (this.lastTime === 0) { // First time update() is called
+      dt = 1;
+    } else {
+      dt = (this.lastTime - currentTime) / 1000; // in seconds
+    }
+    this.lastTime = currentTime;
+  }
+  if (typeof dt !== 'number') {
```

Comment:
Maybe add the condition not to be 0 here? `if (typeof dt !== 'number' || dt === 0)`?